frontend/view-edit-events/event-home/view-edit-event-materials.component.tsx

```diff
@@ -1,42 +1,168 @@
+import { cloneDeep } from "lodash-es";
 import React from "react";
+import AddEditMaterials from "../../add-leads/add-edit-materials.component";
+import { GrowlType, showGrowl } from "../../growls/growls.component";
 import BasicTableReport from "../../reports/shared/basic-table-report.component";
+import easyFetch from "../../util/easy-fetch";
+import { handlePromiseError } from "../../util/error-helpers";
+import Modal from "../../util/modal.component";
 import { SingleEvent } from "../view-event.component";
+import EventMaterial from "./event-material.component";
 import EventSection from "./event-section.component";
+import { useCss } from "kremling";
 
 export default function ViewEditEventMaterials(
   props: ViewEditEventMaterialsProps
 ) {
+  const scope = useCss(css);
+  const [isEditing, setIsEditing] = React.useState(false);
+  const [isSaving, setIsSaving] = React.useState(false);
+  const [materialsDistributed, setMaterialsDistributed] = React.useState([]);
+  const [materials, setMaterials] = React.useState([]);
+  const [showMaterials, setShowMaterials] = React.useState(false);
+
+  React.useEffect(() => {
+    if (!isEditing) {
+      setMaterialsDistributed(cloneDeep(props.event.materialsDistributed));
+    }
+  }, [isEditing, props.event.materialsDistributed]);
```

Comment:
This is a nice usage of an effect. I had to think about it for a while and thought it was unnecessary at first but now see why it's required - we need to update the `materialsDistributed` state whenever `props.event.materialsDistributed` changes, which could happen when you save new materials.

Nicely done in including the correct conditions array for the effect!